File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        idx = 1
        k = 1
        len_nums = len(nums)
        check = 0

        if len(nums) == 1:
            return k

        before_value = nums[0]

        for i in range(1, len_nums):
            if before_value == nums[i]:
                check += 1
            if before_value != nums[i]:
                before_value = nums[i]
                nums[idx] = nums[i]
                idx += 1
                k += 1
        
        if k == 1:
            if check == 0:
                return len_nums
            else:
                return 1
            
        return k

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int left = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[left]) {
                continue;
            } else {
                left++;
                int temp = nums[left];
                nums[left] = nums[i];
                nums[i] = temp;
            }
        }
        return left + 1;
    }
}
